mp2-common/src/eth.rs

```diff
@@ -281,79 +570,591 @@ fn from_rpc_header_to_consensus(h: &alloy::rpc::types::Header) -> alloy::consens
+        // then check the key and make sure it's equal to the RLP encoding of the index
+        let key_header = rlp::Rlp::new(list_buff);
+        assert!(!key_header.is_list());
+        // key is RLP( compact ( RLP(index)))
+        let key_info = key_header.payload_info()?;
+        let compact_key = &list_buff[key_info.header_len..key_info.header_len + key_info.value_len];
+        let decoded_key = rlp::encode(&nibbles_to_bytes(
+            Nibbles::from_compact(compact_key).nibbles(),
+        ));
+        assert_eq!(decoded_key, &mpt_key.rlp_bytes().to_vec(),);
+
+        // then check if the value portion fits what we tested above
+        // value is RLP ( RLP(status, etc...))
+        let outer_value_min = top_info.header_len + key_info.header_len + key_info.value_len;
+        let outer_value_buff = &mpt_node[outer_value_min..];
+        let outer_value_state = rlp::Rlp::new(outer_value_buff);
+        assert!(!outer_value_state.is_list());
+        let outer_payload = outer_value_state.payload_info()?;
+        let inner_value_min = outer_value_min + outer_payload.header_len;
+        let inner_value_buff = &mpt_node[inner_value_min..];
+        assert_eq!(rlp_encoding, inner_value_buff);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_receipt_query() -> Result<()> {
+        let rpc = ProviderBuilder::new()
+            .with_recommended_fillers()
+            .on_anvil_with_wallet_and_config(|anvil| Anvil::block_time(anvil, 1));
+
+        // Make a contract that emits events so we can pick up on them
+        sol! {
+            #[allow(missing_docs)]
+        // solc v0.8.26; solc Counter.sol --via-ir --optimize --bin
+        #[sol(rpc, abi, bytecode="6080604052348015600e575f80fd5b506102288061001c5f395ff3fe608060405234801561000f575f80fd5b506004361061004a575f3560e01c8063488814e01461004e5780638381f58a14610058578063d09de08a14610076578063db73227914610080575b5f80fd5b61005661008a565b005b6100606100f8565b60405161006d9190610165565b60405180910390f35b61007e6100fd565b005b610088610115565b005b5f547fdcd9c7fa0342f01013bd0bf2bec103a81936162dcebd1f0c38b1d4164c17e0fc60405160405180910390a26100c06100fd565b5f547fdcd9c7fa0342f01013bd0bf2bec103a81936162dcebd1f0c38b1d4164c17e0fc60405160405180910390a26100f66100fd565b565b5f5481565b5f8081548092919061010e906101ab565b9190505550565b5f547fdcd9c7fa0342f01013bd0bf2bec103a81936162dcebd1f0c38b1d4164c17e0fc60405160405180910390a261014b6100fd565b565b5f819050919050565b61015f8161014d565b82525050565b5f6020820190506101785f830184610156565b92915050565b7f4e487b71000000000000000000000000000000000000000000000000000000005f52601160045260245ffd5b5f6101b58261014d565b91507fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff82036101e7576101e661017e565b5b60018201905091905056fea26469706673582212202787ca0f2ea71e118bc4d1bf239cde5ec4730aeb35a404c44e6c9d587316418564736f6c634300081a0033")]
+        contract EventEmitter {
+            uint256 public number;
+            event testEvent(uint256 indexed num);
```

Comment:
Could you test with the max number of topics and at least 64B of data as well in the data field ?

Reply:
Will do